buf.yaml

```diff
@@ -0,0 +1,25 @@
+version: v1beta1
+build:
+  roots:
+    - .
+  excludes:
+    - vendor
+lint:
+  use:
+    - BASIC
+    - FILE_LOWER_SNAKE_CASE
+  except:
+    - ENUM_NO_ALLOW_ALIAS
+    - IMPORT_NO_PUBLIC
+    - PACKAGE_AFFINITY
+    - PACKAGE_DIRECTORY_MATCH
+    - PACKAGE_SAME_DIRECTORY
+  ignore_only:
+    FIELD_LOWER_SNAKE_CASE:
+      - plugins/base/proto/base.proto
+      - plugins/device/proto/device.proto
+      - plugins/drivers/proto/driver.proto
+      - plugins/shared/hclspec/hcl_spec.proto
```

Comment:
While `buf` looks pretty exciting, it seems like we're not actually able to take advantage of much of it. It incurs more configuration overhead and introduces another build dependency.

Are the remaining linters impactful enough to outweigh the costs?

Will future protobufs in Nomad be able to take advantage of more of `buf`s features? If that's the case it'd be nice to see them so we can get a better idea of `buf`s impact.

Are other teams at HashiCorp using `buf`?

Reply:
I don't have answers to these questions, so I will just close the PR.

Thanks for reviewing!